### PR TITLE
feat(hooks): useResizeObserver custom sizes

### PR DESCRIPTION
**Pull request by eddysims**
## Motivations

We wanted a better way to use `useResizeObserver` that would allow for some default comparisons, as well as allow the component to set its own breakpoints.

## Changes

`useResizeObserver` will now return the breakpoint size as well as exact size of the component.

### Added

- Custom breakpoints to `useResizeObserver`

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Page/Page.tsx [hunk 1]
```diff
@@ -63,14 +64,15 @@ export function Page({
...
+  const [
+    titleBarRef,
+    { width: titleBarWidth = defaultSizes.large },
+  ] = useResizeObserver<HTMLDivElement>();
 
   const titleBarClasses = classnames(styles.titleBar, {
-    [styles.small]: titleBarWidth && titleBarWidth > 265,
-    [styles.medium]: titleBarWidth && titleBarWidth > 500,
-    [styles.large]: titleBarWidth && titleBarWidth > 750,
+    [styles.small]: titleBarWidth > defaultSizes.smaller,
```
Comment (seanhealy) [inline packages/components/src/Page/Page.tsx:73]: I wonder if exporting `defaultSizes` as `Sizes` or something but be nicer here… I'm not sure why but this feels a bit odd to me… Perhaps I need to think more, also happy to chat.

File: packages/hooks/src/useResizeObserver.ts [hunk 2]
```diff
@@ -4,21 +4,74 @@ import { useMemo, useState } from "react";
...
+  larger: 1024,
+};
+
 interface ObservedSize {
   width: number | undefined;
   height: number | undefined;
 }
 
+interface ResizeObserverProps {
+  width?: object;
```
Comment (seanhealy) [inline packages/hooks/src/useResizeObserver.ts:21]: What do you think of `widths` instead?

File: packages/hooks/src/useResizeObserver.mdx [hunk 3]
```diff
@@ -22,18 +22,30 @@ import { useResizeObserver } from "@jobber/hooks";
...
+      ref,
+      { width, exactWidth, exactHeight, height }
+    ] = useResizeObserver();
     return (
       <div ref={ref}>
-        <Card title={title}>
+        <Card title={`Check out my àçƈéñŦ`} accent={getAccent()}>
           <Content>
             <Text>Width: {width}</Text>
             <Text>Height: {height}</Text>
```
Comment (seanhealy) [inline packages/hooks/src/useResizeObserver.mdx:34]: What do you think of either taking these out or changing the label to `Width Step` and `Height Step`?

File: packages/hooks/src/useResizeObserver.mdx [hunk 4]
```diff
@@ -47,3 +59,56 @@ to represent the `ref`. In the example below, we pass it the type of
...
+      large: 768
+    };
+    const [ref, { width, exactWidth }] = useResizeObserver({
+      width: customWidths
+    });
+    return (
+      <div ref={ref}>
+        <Card title={`Width: ${getCurrentWidth()}`} accent={getAccent()}>
+          <Content>
+            <Text>Width: {width}</Text>
```
Comment (seanhealy) [inline packages/hooks/src/useResizeObserver.mdx:99]: Same here.

File: src/gatsby-theme-docz/components/Playground/index.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
+/* eslint-disable react/prop-types */
```
Comment (seanhealy) [inline src/gatsby-theme-docz/components/Playground/index.js:1]: I'm assuming most of this is direct from the main Docz theme?

Comment (eddysims) [reply src/gatsby-theme-docz/components/Playground/index.js:1]: Yep, only thing i really changed in here was adding the `wrapper` class and changed the margin in the styles.

**Comment by eddysims** [comment]
@seanhealy all comments addressed! back to you

File: packages/hooks/src/useResizeObserver.mdx [hunk 6]
```diff
@@ -47,3 +59,56 @@ to represent the `ref`. In the example below, we pass it the type of
...
+
+export function Sizes() {
+  return <code>{JSON.stringify(Breakpoints)}</code>;
+}
+
+<Sizes />
+
+## Custom Sizes
+
+`useResizeObserver` can take a custom `width` or `height` object if you do not
```
Comment (seanhealy) [inline packages/hooks/src/useResizeObserver.mdx:82]:
```suggestion
`useResizeObserver` can take a custom `widths` or `heights` object if you do not
```

**Pull request merged**